man/chatoob.1
.TH CHATOOB 1 "05 February 2012" "chatoob 0\&.a"
.SH NAME
chatoob
.SH SYNOPSIS
.B chatoob
[\-dqv] [\-b \fIbackends\fR] [\-cnfs] [\fIcommand\fR [\fIarguments\fR..]]
.br
.B chatoob
[\-\-help] [\-\-version]

.SH DESCRIPTION
.LP

Console application allowing to chat with contacts on various websites.

.SS Supported websites:
* aum (“Adopte un mec” french dating website)
.SH CHATOOB COMMANDS
.TP
\fBlist\fR
.br
List all contacts.
.TP
\fBmessages\fR
.br
Get messages.
.TP
\fBsend\fR \fICONTACT\fR \fIMESSAGE\fR
.br
Send a message to the specified contact.
.SH WEBOOB COMMANDS
.TP
\fBbackends\fR [\fIACTION\fR] [\fIBACKEND_NAME\fR]...
.br
Select used backends.
.br

.br
ACTION is one of the following (default: list):
.br
* enable    enable given backends
.br
* disable   disable given backends
.br
* only      enable given backends and disable the others
.br
* list      display enabled and available backends
.br
* add       add a backend
.br
* register  register a new account on a website
.br
* edit      edit a backend
.br
* remove    remove a backend
.TP
\fBcd\fR [\fIPATH\fR]
.br
Follow a path.
.br
If empty, return home.
.TP
\fBcondition\fR [\fIEXPRESSION\fR | off]
.br
If an argument is given, set the condition expression used to filter the results.
.br
If the "off" value is given, conditional filtering is disabled.
.br

.br
If no argument is given, print the current condition expression.
.TP
\fBcount\fR [\fINUMBER\fR | off]
.br
If an argument is given, set the maximum number of results fetched.
.br
NUMBER must be at least 1.
.br
"off" value disables counting, and allows infinite searches.
.br

.br
If no argument is given, print the current count value.
.TP
\fBformatter\fR [list | \fIFORMATTER\fR [\fICOMMAND\fR] | option \fIOPTION_NAME\fR [on | off]]
.br
If a FORMATTER is given, set the formatter to use.
.br
You can add a COMMAND to apply the formatter change only to
.br
a given command.
.br

.br
If the argument is "list", print the available formatters.
.br

.br
If the argument is "option", set the formatter options.
.br
Valid options are: header, keys.
.br
If on/off value is given, set the value of the option.
.br
If not, print the current value for the option.
.br

.br
If no argument is given, print the current formatter.
.TP
\fBinspect\fR \fIBACKEND_NAME\fR
.br
Display the HTML string of the current page of the specified backend's browser.
.br

.br
If webkit_mechanize_browser Python module is installed, HTML is displayed in a WebKit GUI.
.TP
\fBlogging\fR [\fILEVEL\fR]
.br
Set logging level.
.br

.br
Availables: debug, info, warning, error.
.br
* quiet is an alias for error
.br
* default is an alias for warning
.TP
\fBls\fR
.br
List objects in current path.
.TP
\fBquit\fR
.br
Quit the application.
.TP
\fBselect\fR [\fIFIELD_NAME\fR]... | "$direct" | "$full"
.br
If an argument is given, set the selected fields.
.br
$direct selects all fields loaded in one http request.
.br
$full selects all fields using as much http requests as necessary.
.br

.br
If no argument is given, print the currently selected fields.

.SH OPTIONS
.TP
\fB\-\-version\fR
show program's version number and exit
.TP
\fB\-h\fR, \fB\-\-help\fR
show this help message and exit
.TP
\fB\-b BACKENDS\fR, \fB\-\-backends=BACKENDS\fR
what backend(s) to enable (comma separated)

.SH LOGGING OPTIONS
.TP
\fB\-d\fR, \fB\-\-debug\fR
display debug messages
.TP
\fB\-q\fR, \fB\-\-quiet\fR
display only error messages
.TP
\fB\-v\fR, \fB\-\-verbose\fR
display info messages
.TP
\fB\-\-logging\-file=LOGGING_FILE\fR
file to save logs
.TP
\fB\-a\fR, \fB\-\-save\-responses\fR
save every response

.SH RESULTS OPTIONS
.TP
\fB\-c CONDITION\fR, \fB\-\-condition=CONDITION\fR
filter result items to display given a boolean expression
.TP
\fB\-n COUNT\fR, \fB\-\-count=COUNT\fR
get a maximum number of results (all backends merged)
.TP
\fB\-s SELECT\fR, \fB\-\-select=SELECT\fR
select result item keys to display (comma separated)

.SH FORMATTING OPTIONS
.TP
\fB\-f FORMATTER\fR, \fB\-\-formatter=FORMATTER\fR
select output formatter (csv, htmltable, multiline, simple, table, webkit)
.TP
\fB\-\-no\-header\fR
do not display header
.TP
\fB\-\-no\-keys\fR
do not display item keys
.TP
\fB\-O OUTFILE\fR, \fB\-\-outfile=OUTFILE\fR
file to export result

.SH COPYRIGHT
Copyright(C) 2010-2011 Christophe Benz
.LP
For full COPYRIGHT see COPYING file with weboob package.
.LP
.RE
.SH FILES
"~/.weboob/backends" 

.SH SEE ALSO
Home page: http://weboob.org/Chatoob


ChangeLog
Weboob 0.a (2012-02-05)
	General
	* New backend: Boursorama (CapBank).
	* New backend: CMB (CapBank).
	* New backend: HSBC (CapBank).
	* New backend: ING (CapBank).
	* New backend: Nolifetv (CapVideo).
	* New backend: RadioFrance (CapVideo, CapRadio) which merges
	  FranceInter and LeMouv backends.
	* Support repositories to manage backends (#747).
	* Support XDG Base Directory Specification (#765).
	* Make CapCollection understandable and usable by humans.

	Console applications
	* Ability to add all supported modules at once.
	* With -a, try to separate saved responses by backend (#661).

	Qt applications
	* Verbose error messages (#775).

	Backend: AuM
	* Display the 'title' field on contact profiles.
	* Get events (baskets, charms and visits).
	* Add city in status message.
	* Add 'Sent with my iPhone' in mail signatures when needed.
	* Display at least one photo url in profile even if there isn't any
	  visible.
	* Add a 'ratio' field.
	* Fix: getting contacts list.
	* Fix: support new API.
	* Fix: ipaddr is not available anymore.
	* Fix: do not crash if there are php errors before json.

	Backend: Batoto
	* Change .com to .net.

	Backend: BNP
	* Fix: website changed (#769).

	Backend: BP
	* Fix: website changed.
	* Fix: open images in memory instead of saving them in temp files.

	Backend: CanalPlus
	* Support https?://www\.canal-?plus\.fr/.*\?vid=(\d+) urls.

	Backend: CreditMutuel
	* Internal transfer implemented.
	* Fix: website changed.

	Backend: Dailymotion
	* Fix: support missing descriptions.

	Backend: DLFP
	* Limit feed entries by date instead of number of results.

	Backend: FranceTelevisions
	* Fix: website changed.

	Backend: Gazelle
	* Handle general errors when website is unavailable.

	Backend: INA
	* Fix: parsing of URLs.

	Backend: Kickass
	* Fix: website changed.

	Backend: LCL
	* Add support for various types of accounts.
	* List 45 days history by default.
	* Fix: website changed.

	Backend: LeFigaro
	* Better cleanup of generic articles (#725).
	* Fix: website changed.

	Backend: Orange
	* Fix: unicode errors.

	Backend: Pastebin
	* Handle user pages.

	Backend: Piratebay
	* Fix: website changed.

	Backend: SocieteGenerale
	* Fix: website changed.

	Backend: Youjizz
	* Fix: video URLs have changed.

	Backend: Youporn
	* Fix: website changed.

	Backend: Youtube
	* Fix: website changed.

	Application: QBoobMsg
	* When a thread is a discussion, display messages linearly.
	* Add a button 'Profile'.

	Application: QHaveSex
	* Add a 'Events' tab.
	* Add a 'Notes' tab for taking notes about a contact.

	Application: QVideoob
	* Fix: crash when there is no thumbnail on a video (#708).

	Application: weboob-config-qt
	* Ability to edit sources.list, update repositories and install
	  modules.

Weboob 0.9 (2011-10-10)
	General
	* New backend: Batoto (CapGallery).
	* New backend: Eatmanga (CapGallery).
	* New backend: FranceTelevisions (CapVideo).
	* New backend: HDS (CapMessages).
	* New backend: Izneo (CapGallery).
	* New backend: Mangafox (CapGallery).
	* New backend: Mangahere (CapGallery).
	* New backend: Mangareader (CapGallery).
	* New backend: Mangatoshokan (CapGallery).
	* New backend: Nova (CapRadio).
	* New backend: PhpBB (CapMessages, CapMessagesReply).
	* New backend: Simplyread.it (CapGallery).
	* New application: boobtracker (CapBugTracker).
	* New script: hds/export.py to export stories from
	  histoires-de-sexe.net into a sqlite database.
	* License change to AGPLv3+.
	* Ability to not save backend passwords. Applications will prompt them
	  when needed.

	Backend: Arte
	* Set duration on video results.

	Backend: AuM
	* Rewrite to use the mobile API.

	Backend: BNPorc
	* Get operation categories.

	Backend: CrAgr
	* Get full history.
	* Optimized operations retrieval for CA Centre.
	* Fix: parsing accounts list.

	Backend: CreditMutuel
	* Maine-Anjou, Basse-Normandie bank added.

	Backend: Dailymotion
	* Fix: parsing duration.
	* Fix: parsing of lives in search results.

	Backend: DLFP
	* Fix: disconnection URL needs to be called with POST instead of GET.
	* Fix: do tests on alpha.linuxfr.org.
	* Fix: relevance on news/diaries.

	Backend: Gazelle
	* Handle login errors.
	* Fix: parsing on several gazelle websites.

	Backend: INA
	* Support videos on www.ina.fr (in addition to boutique.ina.fr).

	Backend: IPInfoDB
	* Fix: no crash when there are no coordinates from server.

	Backend: Kickass
	* Fix: parsing of search results (#663).

	Backend: MediaWiki
	* Handle errors from API.

	Backend: MeteoFrance
	* Fix: website changes.

	Backend: Orange
	* Fix: handling of pages.

	Backend: PirateBay
	* Fix: parsing of search results.

	Backend: Redmine
	* Now implements CapBugTracker (#684).

	Backend: Transilien
	* Supports the roadmap feature (#681).

	Backend: Youtube
	* Support more URLs.
	* Support authentication.
	* Fix: website changes.

	Application: boobank
	* Add the 'qif' formatter.

	Application: boobank-munin
	* When handling an incorrect password error, disable backend to
	  prevent your bank to disable your account (for example with BNP).

	Application: pastoob
	* Ability to pipe output.

	Application: traveloob
	* New command 'roadmap' to get the roadmap between to stations. It's
	  also possible to give a departure or an arrival time (#681).

	Application: videoob
	* Do not run player in background (#664).
	* Command 'download' supports mms streams.

	Application: weboorrents
	* The given filename to command 'getfile' is optional (#683).

Weboob 0.8 (2011-05-08)
	General
	* New backend: E-Hentai (CapGallery).
	* New backend: FranceInter (CapRadio).
	* New backend: LeMouv (CapRadio).
	* New backend: PasteALaCon (CapPaste).
	* New backend: PasteBin (CapPaste).
	* New application: Boobathon (CapContent).
	* New application: Galleroob (CapGallery).
	* New application: Pastoob (CapPaste).
	* Fix lot of unicode problems.
	* License changed to AGPLv3+.
	* FreeBSD support.

	Repl applications

weboob/applications/chatoob/chatoob.py
# -*- coding: utf-8 -*-

# Copyright(C) 2010-2011 Christophe Benz
#
# This file is part of weboob.
#
# weboob is free software: you can redistribute it and/or modify
# it under the terms of the GNU Affero General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# weboob is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
# GNU Affero General Public License for more details.
#
# You should have received a copy of the GNU Affero General Public License
# along with weboob. If not, see <http://www.gnu.org/licenses/>.


import logging

from weboob.tools.application.repl import ReplApplication
from weboob.capabilities.chat import ICapChat
from weboob.capabilities.contact import ICapContact, Contact


__all__ = ['Chatoob']


class Chatoob(ReplApplication):
    APPNAME = 'chatoob'
    VERSION = '0.b'
    COPYRIGHT = 'Copyright(C) 2010-2011 Christophe Benz'
    DESCRIPTION = 'Console application allowing to chat with contacts on various websites.'
    CAPS = ICapChat

    def on_new_chat_message(self, message):
        print 'on_new_chat_message: %s' % message

    def do_list(self, line):
        """
        list

        List all contacts.
        """
        for backend, contact in self.do('iter_contacts', status=Contact.STATUS_ONLINE, caps=ICapContact):
            self.format(contact)
        self.flush()

    def do_messages(self, line):
        """
        messages

        Get messages.
        """
        for backend, message in self.do('iter_chat_messages'):
            self.format(message)
        self.flush()

    def do_send(self, line):
        """
        send CONTACT MESSAGE

        Send a message to the specified contact.
        """
        _id, message = self.parse_command_args(line, 2, 2)
        for backend, result in self.do('send_chat_message', _id, message):
            if not result:
                logging.error(u'Failed to send message to contact id="%s" on backend "%s"' % (_id, backend.name))
